external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/StringEventDataScheme.java

```diff
@@ -44,21 +40,23 @@
 public class StringEventDataScheme implements IEventDataScheme {
 
   private static final long serialVersionUID = 1L;
+  private static final Logger logger = LoggerFactory.getLogger(StringEventDataScheme.class);
 
   @Override
-  public List<Object> deserialize(Message message) {
+  public List<Object> deserialize(EventData eventData) {
     final List<Object> fieldContents = new ArrayList<Object>();
-
-    for (Section section : message.getPayload()) {
-      if (section instanceof Data) {
-        Data data = (Data) section;
-        fieldContents.add(new String(data.getValue().getArray()));
-      } else if (section instanceof AmqpValue) {
-        AmqpValue amqpValue = (AmqpValue) section;
-        fieldContents.add(amqpValue.getValue().toString());
+    String messageData = "";
+    if(eventData.getBytes()!=null)
+      messageData = new String (eventData.getBytes());
+    else if(eventData.getObject()!=null){
+      try{
+        messageData = new String(Serializedeserializeutil.serialize(eventData.getObject()),Charset.defaultCharset());
```

Comment:
>new String(Serializedeserializeutil.serialize(eventData.getObject()),Charset.defaultCharset()); [](start = 22, length = 95)

eventData.getObject() would return java type of the corresponding Amqp object (if there is no direct java type mapping available - you would see a proton-j type here).
If the type is a primitive data type - simply return .tostring().
else - follow the serialization approach.